controller.go

```diff
@@ -362,7 +366,7 @@ func (c *Controller) syncHandler(key string) error {
 
 		ops := metav1.GetOptions{}
 		name := service.Name + "-tunnel"
-		found, err := tunnels.Get(name, ops)
+		found, err := tunnels.Get(context.Background(), name, ops)
```

Comment:
This is part of upgrading to k8s 1.18. but was that required for the PR? (Just curious)

Reply:
Yes all these interfaces are updated with first parameter being context in k8s 0.18. Hence required.